### PR TITLE
feat/new_vocalfusion

### PR DESCRIPTION
**Pull request by builderjer**
This updates the script to check for a Rpi5 and apply the required overlays.

### CODE REVIEW

File: ovos-i2csound [hunk 1]
```diff
@@ -38,8 +47,21 @@ detect_i2c_device() {
...
+# Gets the board version
+get_board_version() {
+    v=$(tr -d '\0' </proc/device-tree/model)
+    for board in "${!boards[@]}"; do
+        b="${boards[$board]}"
+        if [[ $v =~ "$b" ]]; then
+            board_result="$board"
+        fi
+    done
+    }
```
Comment (BohdanBuinich) [inline ovos-i2csound]: MINOR:

```suggestion
# Gets the board version
get_board_version() {
    v=$(tr -d '\0' </proc/device-tree/model)
    for board in "${!boards[@]}"; do
        b="${boards[$board]}"
        if [[ $v =~ "$b" ]]; then
            board_result="$board"
            break # Exit the loop once the match is found.
        fi
    done
}
```

**Review by BohdanBuinich — APPROVED** [review]

**Comment by j1nx** [comment]
Not tested as of yet but basedon the code it is simple enough to just accept & merge.

So, all looks good to me.

**Review by j1nx — APPROVED** [review]

File: ovos-i2csound [hunk 2]
```diff
@@ -159,7 +189,11 @@ main() {
     if [[ ${detection_results[TAS5806]} == true ]] ; then
         echo "Installing and configuring SJ-201 HAT"
         # Initializing XMOS xvf3510
-        dtoverlay xvf3510
```
Comment (NeonDaniel) [inline ovos-i2csound:162]: This breaks compat with the overlay used for kernels 5.10 to 6.6. Is this intentional or should there be a check here for which overlay exists?

Comment (builderjer) [reply ovos-i2csound:162]: Not intentional. Let me add a check to that.

**Pull request merged**